src/components/pages/OnboardPage.tsx

```diff
@@ -64,17 +60,28 @@ const OnboardPage = ({ unmountSelf }: IOnboardPageProps) => {
           </h1>
         </div>
         <div className="mt-10 flex flex-col">
-          <div className="flex flex-col gap-4" onClick={moveToMain}>
-            {mockPorts.map((port) => {
-              return (
+          <div className="flex flex-col gap-4">
+            {activePorts ? (
+              activePorts.map((port, index) => (
                 <SerialPortOption
                   key={port.name}
                   name={port.name}
                   state={port.state}
-                  connection_error={"Access Denied."}
+                  isActive={activeNode === index}
+                  onSelection={() => {
+                    setActiveNode(index);
+                    setTimeout(() => {
+                      unmountSelf();
+                    }, 1500);
+                  }}
                 />
-              );
-            })}
+              ))
+            ) : (
+              <p className="text-base leading-6 font-normal text-gray-500 pl-40 pr-40 text-center">
+                {" "}
+                No ports detected.{" "}
```

Comment:
This isn't actually showing up when no nodes are plugged in.
![image](https://user-images.githubusercontent.com/57198797/217107255-72cec142-2f2c-4106-a159-275cc0ee4307.png)

Reply:
I believe this is happening because the map isn't null but there are no values. Should now be fixed.